walk/walk.go

```diff
@@ -350,105 +518,26 @@ func findGenFiles(wc *walkConfig, f *rule.File) []string {
 
 	var genFiles []string
 	for _, s := range strs {
-		if !wc.isExcluded(path.Join(f.Pkg, s)) {
+		if !wc.isExcludedFile(path.Join(f.Pkg, s)) {
 			genFiles = append(genFiles, s)
 		}
 	}
 	return genFiles
 }
 
-func shouldFollow(wc *walkConfig, dir, rel string, ent fs.DirEntry) bool {
+func resolveFileInfo(wc *walkConfig, dir, rel string, ent fs.DirEntry) fs.DirEntry {
```

Comment:
Was there a reason this was reverted? Was there an actual bug being fixed (and should I test+fix that before this PR is merged)?

Reply:
Ah, I didn't intend to revert #2054. This was a rebasing mistake.

Looking at it closely though, I think it is better to go back to the original behavior of returning the resolved `fs.DirEntry`. I've renamed, documented, and added a test.

I believe the old code has a bug. It calls `os.ReadDir` to get a list of entries, then `IsDir()` to check whether something is a subdirectory, then `shouldFollow`. `IsDir` returns false on a symbolic link though, so this classifies a followable symbolic link to a directory as a regular file. We need to resolve the symbolic link first before calling `IsDir`.

(Technically we should call `filepath.EvalSymlinks` too before calling `os.Stat` in case there's a chain of symbolic links. I haven't changed that here though.)